### PR TITLE
Add app emoji support

### PR DESCRIPTION
**Pull request by VelvetToroyashi**

### CODE REVIEW

File: DSharpPlus/Clients/DiscordClient.cs [hunk 1]
```diff
@@ -915,6 +980,8 @@ public IAsyncEnumerable<DiscordGuild> GetGuildsAfterAsync(ulong after, int limit
         while (remaining > 0 && lastCount is > 0 and 100);
     }
 
+
+
```
Comment (akiraveliara) [inline DSharpPlus/Clients/DiscordClient.cs:984]: extra newlines

**Review by akiraveliara — CHANGES_REQUESTED** [review]

File: DSharpPlus/Net/Rest/DiscordApiClient.cs [hunk 2]
```diff
@@ -6324,6 +6324,135 @@ internal async ValueTask<GatewayInfo> GetGatewayInfoAsync()
     }
     #endregion
 
+    public async ValueTask<DiscordEmoji> CreateApplicationEmojiAsync(ulong applicationID, string name, string image)
+    {
+        string route = $"{Endpoints.APPLICATIONS}/{applicationID}/{Endpoints.EMOJIS}";
+        string url = $"{Endpoints.APPLICATIONS}/{applicationID}/{Endpoints.EMOJIS}";
+
+        RestGuildCreatePayload pld = new()
```
[the remark this answers is not in the document]
Comment (Plerx2493) [reply DSharpPlus/Net/Rest/DiscordApiClient.cs:6332]: "tested"..

File: DSharpPlus/Net/Abstractions/Rest/RestGuildPayloads.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ internal class RestGuildCreatePayload
     [JsonProperty("region", NullValueHandling = NullValueHandling.Ignore)]
     public string RegionId { get; set; }
 
-    [JsonProperty("icon", NullValueHandling = NullValueHandling.Include)]
+    [JsonProperty("image", NullValueHandling = NullValueHandling.Include)]
```
[the remark this answers is not in the document]
Comment (VelvetToroyashi) [reply DSharpPlus/Net/Abstractions/Rest/RestGuildPayloads.cs:24]: But it's also...wrong? Docs say image

File: DSharpPlus/Net/Rest/DiscordApiClient.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (VelvetToroyashi) [reply DSharpPlus/Net/Rest/DiscordApiClient.cs:6332]: It has been tested yes

File: DSharpPlus/Net/Abstractions/Rest/RestGuildPayloads.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (akiraveliara) [reply DSharpPlus/Net/Abstractions/Rest/RestGuildPayloads.cs:24]: no, no they don't: https://discord.com/developers/docs/resources/guild#create-guild

[the remark this answers is not in the document]
Comment (VelvetToroyashi) [reply DSharpPlus/Net/Abstractions/Rest/RestGuildPayloads.cs:24]: Misread the class name :explosion:

File: DSharpPlus/Net/Rest/DiscordApiClient.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Plerx2493) [reply DSharpPlus/Net/Rest/DiscordApiClient.cs:6332]: Well even if it works, i dont like reusing an unrelated class for smth like this

**Comment by VelvetToroyashi** [comment]
Relevant PR, fwiw: https://github.com/discord/discord-api-docs/pull/7010

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**